Add feature flag for design system

minhngocd opened the pull request:
We are replicating the process we took with Whitehall to use user permissions as a feature flag to transition to the new design system. This will help us implement changes without breaking existing tests and flows while we build out new design system components. 

For testing locally, add `preview_design_system` permission to your local user. 
```
u = User.last
u.permissions << "preview_design_system"
u.save
```

https://trello.com/c/764XzLue/3419-design-system-add-new-document

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

minhngocd committed:
Add user permissions for design transition feature flag

We are replicating the process we took with Whitehall to use user permissions as a feature flag to transition to the new design system.

minhngocd committed:
Feature flag the document controller so it only shows design system or legacy based on user permissions

This will help us implement changes without breaking existing tests and flows while we build out new design system components

minhngocd commented on app/views/documents/new_design_system.html.erb:
```diff
@@ -0,0 +1,31 @@
+
+      <%= render partial: "shared/preview_govspeak", locals: {attachments: @document.attachments} %>
+
+      <%= render partial: "metadata_fields/#{params[:document_type_slug].underscore}", locals: { f: f } %>
```
this is part of existing code - if we're removing the legacy view, this will also go away.

minhngocd committed:
Suppress validation tests for new design system

The new design system can't handle validation messages at the moment, so we are putting a hold on the test expectations until we implement it later.

lauraghiorghisor-tw commented on app/views/documents/new_design_system.html.erb:
```diff
@@ -0,0 +1,31 @@
+<%= content_for :page_title, "New #{current_format.title.singularize}" %>
```
This is all just the same as the old `new` view for now, right?

minhngocd replied:
yes - there's the next PR that does the changing - it's going to be a chunky one so this one is split out on its own for now

lauraghiorghisor-tw reviewed (approved):
LGTM

minhngocd committed:
Do not rendering existing error messages for design system

The method for generating document_error_messages modifies the error messages and make them html safe, which breaks the new layout in the design system.
Before this change, the layout markup breaks after trying to save the form.
The change keeps for layout intact. we will need to think about how we do error message validation for design system at a later point.
